### DISCUSSION TITLE
Burn After Reading Number Counter

### DISCUSSION THREAD

**Discussion by chrish112** [PrivateBin/PrivateBin#1382] [Q&A] [answered]
### Describe the problem/question

Question:
Is it possible to add a Burn After Reading Number Count? 
I have had a few corporate customers complain that the burn after reading links we send them do not work and I have worked with a handful to find user error. After a discussion with some team members we think this other portion of people complaining has to do with email scanning software that is accessing the links or scanning the links for malicious files and therefore burns the link prior to the end-user receiving the link.

I wish to specify that this request is to simply add a counter to how many times the link can be accessed and not to destroy the expiration timer.

If this is not something that is acceptable to roll out in a future update or will be planned for future rollouts may I know what code to input and where? 
This is for PrivateBin v1.7.3

### Did you use the FAQ section?

- [X] Yes, I have read [the FAQ](https://github.com/PrivateBin/PrivateBin/wiki/FAQ) and I found no solution/answer there.

### What you did?

This is a question.

### Server OS

Ubuntu 22 Server

### Webserver

Apache2

### PrivateBin version

v1.7.3

### Issue reproducibility

No, I cannot reproduce it on https://privatebin.net.

**Comment by elrido** [comment, marked as answer]
> Is it possible to add a Burn After Reading Number Count?

No, and maintainers discussed this several times in the past of why introducing such a feature comes with too many risks (to the users) to be worth it.

> After a discussion with some team members we think this other portion of people complaining has to do with email scanning software that is accessing the links or scanning the links for malicious files and therefore burns the link prior to the end-user receiving the link.

That is an actual problem! (services, incl. social media ones, that open the links in a simulated browser, triggering the deletion)

We have long had mitigations in PrivateBin against that, which kept getting circumvented. This is why we eventually saw ourselves forced to introduce the extra human interaction of the dialog we discussed in #1381. It seems to solve this issue, as long as these services don't start to randomly click on buttons.

I would recommend that you explain to your corporate customers that either they "endure" the extra confirmation dialog or accept that pastes get deleted prematurely, if shared through such tools.

If you do find evidence that a certain service is bypassing the dialog and still triggers the deletion, please raise an issue and name and shame the service in question, so we can attempt to reproduce it (if it's a public service) or PrivateBin users at least know not to use those for burn-after-reading pastes.

Finally, of course you can also send links with a very short expiration time, for example a few hours. That way they can get accessed several times, but you can still rely on them not being available forever.
